src/routes/validity.js

```diff
@@ -0,0 +1,67 @@
+const { getContractErrors } = require('../db/nodeDb');
+const { isTxIdValid, getContractState } = require('../common');
+const LRUCache = require('../lruCache');
+
+const validityCache = new LRUCache(1000);
+
+module.exports = async (ctx) => {
+  const id = ctx.query.id;
+  const contractId = ctx.query.contractId;
+  const { nodeDb } = ctx;
+  let response = {};
+
+  if (!id || !contractId) {
+    ctx.throw(400, 'Id and contract id must be provided.');
+  }
+
+  if (!isTxIdValid(id)) {
+    ctx.throw(400, 'Interaction id not valid.');
+  }
+
+  if (!isTxIdValid(contractId)) {
+    ctx.throw(400, 'Contract id not valid.');
+  }
+
+  try {
+    const contractValidity = validityCache.get(contractId);
```

Comment:
hmm..tbh I though about storing a map from interaction_id -> result, not contract_id -> full_validity

if we're gonna store full contract validity, reduce the capacity to sth like 20...as full contract validity and error messages can be BIG.

Reply:
ok, let's leave the current version, but please reduce the capacity to 20.

Reply:
ok, done